lib/fauna/client.rb

```diff
@@ -171,6 +172,17 @@ def execute(action, path, query = nil, data = nil)
           response_raw, response_content, response.status, response.headers,
           start_time, end_time)
 
+
+      if response.headers.key?('X-Txn-Time')
+        time = response.headers['X-Txn-Time'].to_i
+
+        if @last_seen.nil?
+          @last_seen = time
+        elsif @last_seen < time
+          @last_seen = time
```

Comment:
There isn't much around concurrency in this driver but, this method may be invoked from multiple threads from the application code.

Reply:
"This client is not thread safe". :P

Reply:
Isn't it? I thought it was.

Reply:
My fault for being vaguely humorous: I was making a joke about adding a comment.

Reply:
Oh. It was a good joke! haha